utility/cli_message_examples_reprex.md

```diff
@@ -157,11 +212,72 @@ missing_list_col_supertbl <- tibble(redcap_data = list(), redcap_metadata = 123)
   as_supertbl()
 make_labelled(missing_list_col_supertbl)
 #> Error in `make_labelled()`:
-#> ✖ You've supplied `<redcap_supertbl[,2]>` for `supertbl` which is not a
-#>   valid value
+#> ✖ You've supplied `<rdcp_spr[,2]>` for `supertbl` which is not a valid
+#>   value
 #> ! `supertbl$redcap_metadata` must be of type 'list'
 #> ℹ `supertbl` must be a REDCapTidieR supertibble, generated using
 #>   `read_redcap()`
+
+# add_skimr_metadata
+
+mtcars %>% add_skimr_metadata()
+#> Error in `add_skimr_metadata()`:
+#> ✖ You've supplied `<df[,11]>` for `supertbl` which is not a valid value
+#> ! Must be of class <redcap_supertbl>
+#> ℹ `supertbl` must be a REDCapTidieR supertibble, generated using
+#>   `read_redcap()`
+
+# write_redcap_xlsx
+
+withr::with_tempdir({
+  dir <- getwd()
+  filepath <- paste0(dir, "/temp.csv")
+  REDCapTidieR:::check_file_exists(file = filepath, overwrite = FALSE)
+})
+
+withr::with_tempdir({
+  dir <- getwd()
+  tempfile <- write.csv(x = mtcars, file = "temp.csv")
+  filepath <- paste0(dir, "/temp.csv")
+  REDCapTidieR:::check_file_exists(file = filepath, overwrite = FALSE)
+})
+#> Error:
+#> ✖ File
+#>   ''/private/var/folders/qc/mmjjyjq50530z9r_7mfqcqfhxkkk67/T/Rtmp9DL6jB/file35a532406eec/temp.csv''
+#>   already exists.
+#> ℹ Set `overwrite = TRUE` to overwrite your file.
+
+write_redcap_xlsx(mtcars, file = "temp.xlsx")
+#> Error in `write_redcap_xlsx()`:
+#> ✖ You've supplied `<df[,11]>` for `supertbl` which is not a valid value
+#> ! Must be of class <redcap_supertbl>
+#> ℹ `supertbl` must be a REDCapTidieR supertibble, generated using
+#>   `read_redcap()`
+
+read_redcap(redcap_uri, classic_token) %>%
+  write_redcap_xlsx(file = "temp.xlsx", add_labelled_column_headers = TRUE)
+#> Warning in read_redcap(redcap_uri, classic_token): ! Multiple values are mapped to the label `A` in field `radio_duplicate_label`
+#> ℹ Consider making the labels for `radio_duplicate_label` unique in your REDCap
+#>   project
+#> Warning: ! The field radio_dtxt_error in data_field_types is a radio field type, however
+#>   it does not have any categories.
+#> Warning: Empty string detected for a given multiple choice label.
+#> Error in `write_redcap_xlsx()`:
+#> ✖ `add_labelled_column_headers` declared TRUE, but no variable labels
+#>   detected.
+#> ℹ Did you run `make_labelled()` on the supertibble?
+
+read_redcap(redcap_uri, classic_token) %>%
+  write_redcap_xlsx(file = "temp.pdf")
+#> Warning in read_redcap(redcap_uri, classic_token): ! Multiple values are mapped to the label `A` in field `radio_duplicate_label`
+#> ℹ Consider making the labels for `radio_duplicate_label` unique in your REDCap
+#>   project
+#> Warning: ! The field radio_dtxt_error in data_field_types is a radio field type, however
+#>   it does not have any categories.
+#> Warning: Empty string detected for a given multiple choice label.
+#> Error in `write_redcap_xlsx()`:
+#> ✖ Invalid file extension supplied to `file`: pdf
```

Comment:
I think this should be a warning not an error. 

What happens if no file extension is given? What if someone provides ".xls"?

Reply:
Using `wb$save()` will execute for pretty much any extension, but the files themselves will be corrupted, produce warnings, or be not be openable. In testing, CSV and XLS save but they open with warnings. docx and pdf will save, but the files won't open. When I tested no file extension it was an empty file.

I guess it's a determination of if we want to fail loudly or silently and let users discover for themselves if their files work. CSV and XLS will be openable, but users will have to convert the files to XLSX to retain the XLSX features we have in our output.